CONTRIBUTING.md

```diff
@@ -63,6 +63,20 @@ To run unit tests or Web Platform Tests against a pull request, add one or more
 | `T-macos` | Unit tests: macOS |
 | `T-windows` | Unit tests: Windows |
 
+## AI contributions
+
+Contributions must not include content generated by large language models or other probabilistic tools, including but not limited to Copilot or ChatGPT. This policy covers code, documentation, pull requests, issues, comments, and any other contributions to the Servo project.
+
+This field is evolving quickly, so we may decide to revise this policy at a later date. For now, we’re taking a cautious approach to these tools due to their effects — both unknown and observed — on project health and maintenance burden. Our rationale is as follows:
+
+**Maintainer burden:** Reviewers depend on contributors to write and test their code before submitting it. We have found that these tools make it easy to generate large amounts of plausible-looking code that the contributor does not understand, is often untested, and does not function properly. This is a drain on the (already limited) time and energy of our reviewers.
+
+**Correctness and security:** Even when code generated by AI tools does seem to function, there is no guarantee that it is correct, and no indication of what security implications it may have. A web browser engine is built to run in hostile execution environments, so all code must take into account potential security issues. Contributors play a large role in considering these issues when creating contributions, something that we cannot trust an AI tool to do.
+
+**Copyright issues:** Publicly available models are trained on copyrighted content, both accidentally and intentionally, and their output often includes that content verbatim. Since the legality of this is uncertain, these contributions could make the project liable for copyright infringement.
+
+**Ethical issues:** AI tools require an unreasonable amount of energy and water to build and operate, their models are built with heavily exploited workers in unacceptable working conditions, and are being used to undermine labor and justify layoffs. These are harms that we do not want to perpetuate, even if only indirectly.
```

Comment:
I would reword this one as a plea to contributors to consider the ethics of their AI use, by:

1. Adding :"We therefore ask contributors to consider these issues before using AI tools to contribute to Servo, especially due to the limit of their current usefulness on a large codebase." 
2. Rephrasing as "It has been reported in the press that certain AI tools require (...), and that their models are built with (...).", and add a few links as examples.